src/ImageSharp/ImageFrame{TPixel}.cs

```diff
@@ -166,6 +167,39 @@ internal ImageFrame(Configuration configuration, ImageFrame<TPixel> source)
             }
         }
 
+        /// <summary>
+        /// Gets the representation of the pixels as a <see cref="Span{T}"/> of contiguous memory
+        /// at row <paramref name="rowIndex"/> beginning from the first pixel on that row.
+        /// </summary>
+        /// <param name="rowIndex">The row.</param>
+        /// <returns>The <see cref="Span{TPixel}"/></returns>
+        public Span<TPixel> GetPixelRowSpan(int rowIndex)
+        {
+            Guard.MustBeGreaterThanOrEqualTo(rowIndex, 0, nameof(rowIndex));
+            Guard.MustBeLessThan(rowIndex, this.Height, nameof(rowIndex));
+
+            return this.PixelBuffer.GetRowSpan(rowIndex);
+        }
+
+        /// <summary>
+        /// Gets the representation of the pixels as a <see cref="Span{T}"/> in the source image's pixel format
+        /// stored in row major order, if the backing buffer is contiguous.
+        /// </summary>
+        /// <param name="span">The <see cref="Span{T}"/>.</param>
+        /// <returns>The <see cref="bool"/>.</returns>
+        public bool TryGetSinglePixelSpan(out Span<TPixel> span)
+        {
+            IMemoryGroup<TPixel> mg = this.GetPixelMemoryGroup();
+            if (mg.Count > 1)
+            {
+                span = default;
+                return false;
```

Comment:
@antonfirsov Is there a test somewhere I can tap into to test the false condition?

Reply:
Yes, you can limit `MemoryAllocator`'s `BufferCapacity` in many ways to ensure the backing memory is fragmented: 
- [like this](https://github.com/SixLabors/ImageSharp/blob/master/tests/ImageSharp.Tests/Memory/Buffer2DTests.cs#L43) (works with `TestMemoryAllocator` only).
- [or like this](https://github.com/SixLabors/ImageSharp/blob/master/tests/ImageSharp.Tests/Formats/Jpg/JpegDecoderTests.cs#L109).